### ISSUE TITLE
commandLine can't add addons outside of OF_ROOT\addons

### ISSUE THREAD

**Issue by mattfelsen** [openframeworks/projectGenerator#23]
I'm trying to set up a project with this sort of structure (it's a slight deviation from [this setup](http://local-projects.github.io/))

PROJECT_ROOT
- Apps
- Docs
- submodules
  - openFrameworks (i.e. OF_ROOT)
  - ofxAddon1
  - ofxAddon2

This is nice because it lets us manage both OF and all the addons as git submodules. What I'm seeing with the commandLine tool is it takes arguments from the addons="" param and tries to append it to $OF_ROOT\addons, which doesn't make sense in this case, even if I pass a full path as the addon param. My verbose output is below.

I'm much newer to VS than Xcode...adding addons to existing projects in Xcode is pretty easy with the plugin, but I'm finding it a pain to add addons to VS that require manual setup of include paths, library paths, library names, etc. so I was hoping the commandLine tool would be able to help with updating existing projects. Does this seem like the right approach?

```
> commandLine.exe /verbose /ofPath=C:\Users\mattfelsen\Documents\Code\Projects\CMA2\submodules\openFrameworks C:\Users\mattfelsen\Documents\Code\Projects\CMA2\Sketches\TestAddon /addons=C:\Users\mattfelsen\Documents\Code\Projects\CMA2\submodules\ofxEyeTribe


[notice ] PG_OF_PATH not set (see help), -o parameter needs to be set (-h for options)
[notice ] C:\Users\mattfelsen\Documents\Code\Projects\CMA2\Sketches\TestAddon does not exist, using 'create' mode
[notice ] setting OF path to: C:\Users\mattfelsen\Documents\Code\Projects\CMA2\submodules\openFrameworks


[notice ] -----------------------------------------------
[notice ] setting up a new project
[notice ] target platform is: vs
[notice ] project path is: C:\Users\mattfelsen\Documents\Code\Projects\CMA2\Sketches\TestAddon
[verbose] loaded filter
[verbose] returning path ../../submodules/openFrameworks/

[notice ] parsing addon: C:\Users\mattfelsen\Documents\Code\Projects\CMA2\submodules\openFrameworks\addons
[verbose] in fromFS, trying src C:\Users\mattfelsen\Documents\Code\Projects\CMA2\submodules\openFrameworks\addons\C:\Users\mattfelsen\Documents\Code\Projects\CMA2\submodules\ofxEyeTribe/src
[ error ] ofDirectory: listDir:() source directory does not exist: ""C:\Users\mattfelsen\Documents\Code\Projects\CMA2\submodules\openFrameworks\addons\C:\Users\mattfelsen\Documents\Code\Projects\CMA2\submodules\ofxEyeTribe/src\""
[verbose] trying get folders recursively C:\Users\mattfelsen\Documents\Code\Projects\CMA2\submodules\openFrameworks\addons\C:\Users\mattfelsen\Documents\Code\Projects\CMA2\submodules\ofxEyeTribe/libs
[ error ] ofDirectory: listDir:() source directory does not exist: ""C:\Users\mattfelsen\Documents\Code\Projects\CMA2\submodules\openFrameworks\addons\C:\Users\mattfelsen\Documents\Code\Projects\CMA2\submodules\ofxEyeTribe/libs\""
[ error ] ofDirectory: listDir:() source directory does not exist: ""C:\Users\mattfelsen\Documents\Code\Projects\CMA2\submodules\openFrameworks\addons\C:\Users\mattfelsen\Documents\Code\Projects\CMA2\submodules\ofxEyeTribe/src\""
[verbose] adding addon include path: ../../../addons\C:\Users\mattfelsen\Documents\Code\Projects\CMA2\submodules\ofxEyeTribe/libs
[verbose] adding addon include path: ../../../addons\C:\Users\mattfelsen\Documents\Code\Projects\CMA2\submodules\ofxEyeTribe/src
[notice ] saving addons.make
[notice ] project created!
[notice ] -----------------------------------------------


1 project created in 0.325195 seconds
```

**Comment by ofZach** [comment]
there's no support at the moment for addons that are not in the addons folder...   I think it shouldn't be super hard to add but it's not on my radar for 0.9 (the gui for this is)

If you wanted to poke at it, I think some thought would have to be put into these specific places: 

https://github.com/openframeworks/openFrameworks/blob/master/addons/ofxProjectGenerator/src/addons/ofAddon.cpp#L404-L409

https://github.com/openframeworks/openFrameworks/blob/master/addons/ofxProjectGenerator/src/addons/ofAddon.cpp#L194

https://github.com/openframeworks/openFrameworks/blob/master/addons/ofxProjectGenerator/src/addons/ofAddon.cpp#L194

https://github.com/openframeworks/projectGenerator/blob/master/commandLine/src/main.cpp#L617-L622

it's possible to imagine some tests in the main.cpp (command line tool) that check if the addon is a single word or an absolute path and do different things.  I haven't really looked into the what the code in ofAddon.cpp is doing....

If you want to give it a shot, go for it !  I am happy to look at this after 0.9 as well.

**Comment by mattfelsen** [comment]
Fixed via #26. Thanks again @arturoc!!
